ValveKeyValue/ValveKeyValue.Test/Helpers/BlockingStream.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.IO;
+using System.Text;
+
+namespace ValveKeyValue.Test.Helpers;
+
+/// <summary>
+///     Example implementation of <see cref="Stream" /> that reads at most <see cref="MaxReadAtOnce" /> characters during a
+///     single <see cref="Read" /> operation, blocking in the process.
+/// </summary>
+internal sealed class BlockingStream : Stream
+{
+    private static readonly byte[] TestData = Encoding.UTF8.GetBytes(
+        @"""test_kv""
+		{
+			""test""	""1337""
+		}"
+    );
+
+    private long _backingPosition;
+
+    internal BlockingStream(int maxReadAtOnce = 1)
```

Comment:
If `maxReadAtOnce` is customisable then `BlockingStream` isn't guaranteed to be blocking.
It looks like it's only guaranteed to be blocking when our `maxReadAtOnce` is smaller than the `StreamReader`'s `bufferSize`, which is also configurable.

It may also be possible that if `StreamReader`'s `bufferSize` is 1 then it will never block. 🤔